sql/api/src/main/java/org/apache/spark/sql/streaming/TimeMode.java

```diff
@@ -31,11 +30,6 @@
 @Evolving
 public class TimeMode {
 
-    /**
-     * Neither timers nor ttl is supported in this mode.
-     */
-    public static final TimeMode None() { return NoTime$.MODULE$; }
```

Comment:
When did we add this? If this is already released out, we can't just remove and break it.

Reply:
We added this on April 7, 2024 as part of TTL support for Streaming State with TransformWithState operator. This operator is a new addition to Spark Structured Streaming, and is planned to be released in Spark 4.1. The API is currently in heavy development and evolving. I think its safe to make this change at this point, given its still unreleased in a Spark Major version.

Reply:
yea not released yet. This is only in the 4.0 branch. 
@sahnib - while we are here - is there a way to make this package private to `sql` too on the Java side ?

Reply:
Java doesn't support package private with specifying ancestor package. It's always limited to the same package and it won't work for our case.
(It might change in recent versions though I haven't heard about such change.)